app/controllers/stormpath/rails/social/linkedin_controller.rb

```diff
@@ -0,0 +1,18 @@
+module Stormpath
+  module Rails
+    module Social
+      class LinkedinController < Stormpath::Rails::Social::SocialController
+        def create
+          begin
+            request = Stormpath::Provider::AccountRequest.new(:linkedin, :code, params[:code])
+            account = Stormpath::Rails::Client.application.get_provider_account(request).account
+            login_the_account(account)
+            respond_with_success
+          rescue InvalidSptokenError, NoSptokenError => error
+            respond_with_error(error)
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
account and request can be separate private methods (the extract method technique). Apply to other controllers.